ResearchKit/Common/ORKBaseGraphView.m

```diff
@@ -0,0 +1,870 @@
+    
+    if (self.emptyLabel) {
+        self.emptyLabel.frame = CGRectMake(ORKGraphLeftPadding, ORKGraphTopPadding, CGRectGetWidth(self.frame) - ORKGraphLeftPadding, CGRectGetHeight(self.frame) - XAxisHeight - ORKGraphTopPadding);
+    }
+    
+    //Scrubber Views
+    self.scrubberLine.frame = CGRectMake(CGRectGetMinX(self.scrubberLine.frame), ORKGraphTopPadding, 1, CGRectGetHeight(self.plotsView.frame));
+    [self updateScrubberLabel];
+    self.scrubberThumbView.frame = CGRectMake(CGRectGetMinX(self.scrubberThumbView.frame), CGRectGetMinY(self.scrubberThumbView.frame), [self scrubberThumbSize].width, [self scrubberThumbSize].height);
+    self.scrubberThumbView.layer.cornerRadius = self.scrubberThumbView.bounds.size.height/2;
+    
+    [_xAxisView layoutSubviews];
+}
+
+- (void)setDefaults {
+    _minimumValue = MAXFLOAT;
+    _maximumValue = -MAXFLOAT;
+}
+
+#pragma mark - Drawing
+
+- (void)refreshGraph {
+    //Clear subviews and sublayers
+    [self.plotsView.layer.sublayers makeObjectsPerformSelector:@selector(removeAllAnimations)];
+    [self.plotsView.layer.sublayers makeObjectsPerformSelector:@selector(removeFromSuperlayer)];
+    
+    [self drawXAxis];
+    [self drawYAxis];
+    
+    [self drawHorizontalReferenceLines];
+    
+    if (self.showsVerticalReferenceLines) {
+        [self drawVerticalReferenceLines];
+    }
+    
+    [self calculateXAxisPoints];
+    
+    [self.dots removeAllObjects];
+    [self.pathLines removeAllObjects];
+    
+    for (int i=0; i<[self numberOfPlots]; i++) {
+        if ([self numberOfPointsinPlot:i] <= 1) {
+            return;
+        } else {
+            [self drawGraphForPlotIndex:i];
+        }
+    }
+    
+    if (!self.hasDataPoint) {
+        [self setupEmptyView];
+    } else {
+        if (self.emptyLabel) {
+            [self.emptyLabel removeFromSuperview];
+        }
+    }
+    
+    [self animateLayersSequentially];
+}
+
+- (void)prepareDataForPlotIndex:(NSInteger)plotIndex {
+    [_dataPoints removeAllObjects];
+    [_yAxisPoints removeAllObjects];
+    _hasDataPoint = NO;
+    for (int i = 0; i<[self numberOfPointsinPlot:plotIndex]; i++) {
+        
+        if ([_dataSource respondsToSelector:@selector(graphView:plot:valueForPointAtIndex:)]) {
+            ORKRangePoint *value = [self.dataSource graphView:self plot:plotIndex valueForPointAtIndex:i];
+            [_dataPoints addObject:value];
+            
+            if (!value.isEmpty){
+                _hasDataPoint = YES;
+            }
+        }
+    }
+    [self.yAxisPoints addObjectsFromArray:[self normalizeCanvasPoints:self.dataPoints forRect:self.plotsView.frame.size]];
+}
+
+- (void)drawGraphForPlotIndex:(NSInteger)plotIndex {
+    [self prepareDataForPlotIndex:plotIndex];
+    [self drawPointCirclesForPlotIndex:plotIndex];
```

Comment:
The point circles should be drawn after the lines (the lines should not be seen behind the circles).